### PR TITLE
Add reporter options

### PR DESCRIPTION
**Pull request by klzns**
As discussed in: https://github.com/KeywordBrain/redbox-react/issues/22 and https://github.com/aulizko/delicate-error-reporter/issues/6#issuecomment-139912921

It would be great if the reporter could get some options as props.

### CODE REVIEW

File: src/development.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export default function catchErrors({ filename, components, imports }) {
-  const [React, ErrorReporter] = imports;
+  const [React, ErrorReporter, ReporterOptions] = imports;
```
Comment (gaearon) [inline src/development.js]: nit: let's make it camel case, i.e. `reporterOptions`

File: src/development.js [hunk 2]
```diff
@@ -22,10 +22,15 @@ export default function catchErrors({ filename, components, imports }) {
         } else {
           console.error(err);
         }
-        return React.createElement(ErrorReporter, {
-          error: err,
-          filename
-        });
+        return React.createElement(ErrorReporter,
```
Comment (gaearon) [inline src/development.js:25]: Let's style it as

``` js
React.createElement(ErrorReporter, Object.assign({}, reporterOptions, {
  error: err,
  filename
}))
```

This way we have less nesting. Also, `|| {}` is not necessary inside `Object.assign`—it skips undefined stuff.

Comment (gaearon) [inline src/development.js:25]: Oh, and it's entirely possible the user doesn't have `Object.assign` polyfill. Please add `babel-plugin-object-assign` to this project's `.babelrc` so it compiles to `_extend` helper call.

You can also set `stage` to be `0` and just say `React.createElement(ErrorReporter, { error: err, filename, ...reporterOptions })`. In fact I'd prefer this.

**Comment by klzns** [comment]
@gaearon thanks for the feedback! I updated the code

**Comment by gaearon** [comment]
Good. Have you tested it? (Does it work as intended with `redbox-react`?)

File: src/development.js [hunk 3]
```diff
@@ -22,10 +22,12 @@ export default function catchErrors({ filename, components, imports }) {
...
+          {
```
Comment (gaearon) [inline src/development.js:26]: Please move this brace to the previous line.

**Comment by klzns** [comment]
@gaearon yes, it does! I submitted a pull request to them here: https://github.com/KeywordBrain/redbox-react/pull/23

Everything works fine :)

**Comment by gaearon** [comment]
Released as 0.2.0.

**Pull request merged**